src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -108,31 +108,36 @@ define([
          */
         _loadData: function (callback) {
             logger.debug(this.id + "._loadData");
-            this._clearValidations();
 
-            //default fetch
-            var refEntity = this.reference.split("/")[1],
-                filters = {},
-                xpath = "//" + refEntity;
+            if (this._contextObj && this._contextObj.getGuid()) {
+                this._clearValidations();
 
-            filters.sort = [[this.sortAttr, this.sortOrder]];
-            if (this.limit > 0) {
-                filters.amount = this.limit;
-            }
-            if (this.constraint) {
-                xpath = "//" + refEntity + this.constraint.replace(/\[%CurrentObject%\]/g, this._contextObj.getGuid());
+                //default fetch
+                var refEntity = this.reference.split("/")[1],
+                    filters = {},
+                    xpath = "//" + refEntity;
+
+                filters.sort = [[this.sortAttr, this.sortOrder]];
+                if (this.limit > 0) {
+                    filters.amount = this.limit;
+                }
+                if (this.constraint) {
+                    xpath = "//" + refEntity + this.constraint.replace(/\[%CurrentObject%\]/g, this._contextObj.getGuid());
+                }
+                mx.data.get({
+                    xpath: xpath,
+                    filter: filters,
+                    callback: lang.hitch(this, function(objs) {
+                        this._fetchData(objs, callback);
+                    }),
+                    error: lang.hitch(this, function(err) {
+                        console.error(this.id + "_loadData get failed: " + err.toString());
+                        callback(null);
```

Comment:
Could `callback` be undefined here? There's guards elsewhere for ReferenceErrors.

Reply:
Fixed